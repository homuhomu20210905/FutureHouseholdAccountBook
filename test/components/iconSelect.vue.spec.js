import { mount, shallowMount } from '@vue/test-utils'
import iConSelect from '@/components/iconSelect.vue'
import Vuetify from 'vuetify'

const vuetify = new Vuetify()

describe('mount', () => {
  test('props check', async () => {
    // const wrapper = mount(iConSelect)
    // await wrapper.setProps({ position: 1 })
    const wrapper = shallowMount(iConSelect, {
      propsData: { position: 1 },
      vuetify,
    })
    expect(wrapper.vm).toBeTruthy()
    expect(wrapper.vm.current).toBe('浪費')
    expect(wrapper.element).toMatchSnapshot()
    // expect(wrapper.emitted('update:choiceStatusId')).toBeTruthy()
    // expect(wrapper.emitted('update:choiceStatusId')[0][0]).toBe(100)

    // await wrapper.setProps({ position: 1 })
    // expect(wrapper.vm).toBeTruthy()
    // expect(wrapper.vm.current).toBe('浪費')
  })
})
