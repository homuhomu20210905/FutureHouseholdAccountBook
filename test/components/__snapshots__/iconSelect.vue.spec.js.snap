// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mount props check 1`] = `
<div>
  <v-chip-group-stub
    activeclass="v-slide-item--active"
    column="true"
    mobilebreakpoint="1264"
    nexticon="$next"
    previcon="$prev"
    tag="div"
  >
    <v-chip-stub
      active="true"
      activeclass="light-blue darken-3"
      closeicon="$delete"
      closelabel="$vuetify.close"
      filtericon="$complete"
      tag="span"
    >
      
      浪費
    
    </v-chip-stub>
    <v-chip-stub
      active="true"
      activeclass="light-blue darken-3"
      closeicon="$delete"
      closelabel="$vuetify.close"
      filtericon="$complete"
      tag="span"
    >
      
      消費
    
    </v-chip-stub>
    <v-chip-stub
      active="true"
      activeclass="light-blue darken-3"
      closeicon="$delete"
      closelabel="$vuetify.close"
      filtericon="$complete"
      tag="span"
    >
      
      投資
    
    </v-chip-stub>
  </v-chip-group-stub>
</div>
`;
